### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Fs00**
I took some time to update all dependencies (some of them were a couple of years old) and made sure that all of them were compatible with each other (for example, auto-value has been left a few versions behind in order to be compatible with auto-value-parcel).
Also updated Gradle, AGP, and replaced legacy support libraries with AndroidX.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,113 +1,113 @@
...
-            }
-        }
-    }
-}
-
-task clean(type: Delete) {
-    delete rootProject.buildDir
-}
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
```
Comment (maniac103) [inline build.gradle]: Whitespace in this file should be kept to reduce the diff.

Comment (Fs00) [reply build.gradle]: I'm really not sure what happened here. The whitespace seems the same to me, maybe line endings changed?

Comment (maniac103) [reply build.gradle]: Probably. A meaningful editor should give a clue ;-)

Comment (Fs00) [reply build.gradle]: Ok, now it should be good to go.

**Comment by Fs00** [comment]
Rebased and ready. You can publish a new version of the bindings once you merge the currently open PRs, I have no other things planned for now :)

**Pull request merged**